### PR TITLE
Fix non-closure of file descriptor

### PR DESCRIPTION
**Pull request by chrisgzf**
I found something to nitpick on just so that I can have a commit under my name in the legendary Russell Saerang's pancake repo.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -105,10 +105,11 @@ def parse(soup):
     while True:
         try:
             print(input(f'Paste the whole page source at {HTML_FN}, and press Enter when you\'re done:\n'))
-            html_doc = ''.join(map(bytes.decode, open(HTML_FN, 'rb').readlines()))
+            with open(HTML_FN, 'r') as f:
```
Comment (RussellDash332) [inline main.py]: If possible, can you retain the file mode at `'rb'`? Some bytes can't be decoded in `'r'` mode 😭

File: main.py [hunk 2]
```diff
@@ -105,10 +105,11 @@ def parse(soup):
     while True:
         try:
             print(input(f'Paste the whole page source at {HTML_FN}, and press Enter when you\'re done:\n'))
-            html_doc = ''.join(map(bytes.decode, open(HTML_FN, 'rb').readlines()))
+            with open(HTML_FN, 'r') as f:
+                html_doc = ''.join(f.readlines())
```
Comment (RussellDash332) [inline main.py]: That also means we still have to use `bytes.decode`

**Review by RussellDash332 — CHANGES_REQUESTED** [review]
Thank you for the PR, Lord Christopher Goh, but I have some issues to address

File: main.py (hunk 2 above)
Comment (chrisgzf) [reply main.py]: Thank you for your valuable and insightful comments! Definitely agree with your suggested changes! Fixed in [0872f78](https://github.com/RussellDash332/pancake/pull/1/commits/0872f7833bfbd33fa5595da0ff2c0b0268698518).

File: main.py (hunk 1 above)
Comment (chrisgzf) [reply main.py]: Thank you for your valuable and insightful comments! Definitely agree with your suggested changes! Fixed in [0872f78](https://github.com/RussellDash332/pancake/pull/1/commits/0872f7833bfbd33fa5595da0ff2c0b0268698518).

**Review by RussellDash332 — APPROVED** [review]
LGTM!

**Pull request merged**